### ISSUE TITLE
Suggestion: add original filename to renaming rule fields

### ISSUE THREAD

**Issue by r0bis** [jlegewie/zotfile#296]
Thank you very much for this fantastic add-on. It is very good.

Would it be difficult to add original filename as a possible field to the attachment renaming, say as `{%o}`. Use case would be this:

I have an article - it consists of several parts and I want to keep them separate; however they all belong to the same article - therefore I want only **one** entry in Zotero. Names could be like this:
```
My_article_name.pdf
Abstract.pdf
Diagram_1.pdf
Diagram_2.pdf
Table_1.pdf
Slide_1.pdf
```

If I had the option to add original filename, it would create a nice series of attachments, all belonging to the same article. All I would have to do after I add the main text (the first list item) of the article  would be to change the rules in the preferences by adding {%o} and then I can add the other attachments. Or if I choose to name the first item simply as `main.pdf` I would not even need to change renaming preferences. I presently achieve similar results via the user input field, but that is a manual entry every time and also creates a whitespace between the name and the user input, which I do not like in filenames. 

I think there could be many use cases when one wants one bibliographic item to refer to, but to have multiple attachments that each may have a different use case (e.g. email just the table to the editor etc.). I am very happy to use user input field, but I thought that adding the original filename field might be an easy and useful addition that expands use of Zotero as an electronic library manager.

With best regards,
Robert

**Comment by jlegewie** [comment]
What is about the `extensions.zotfile.disable_renaming` option? Read more here: http://zotfile.com/#hidden-options

**Comment by r0bis** [comment]
Thanks, I am aware one can disable renaming, but renaming is very useful. The idea is to use filename as one field - for example with that I have an article consisting of main item, abstract, 3 diagrams - it would be great to have Zotfile rename them like:
`
- My_article_2017_main.pdf  
- My_article_2017_abstract.pdf  
- My_article_2017_diagram01.pdf  
- My_article_2017_diagram02.pdf  
- My_article_2017_diagram03.pdf 
`
So the name fields 1-3 - author_articleName_year - are taken from Zotero metadata, but the fourth is taken from the original filename. This would provide me with parts of my article that have consistent names and allow to send any part I have to colleagues/publishers easily. I know it is just a convenience function, but it could be very useful if I wanted to use the original filename as a part of attachment name.   It could be very usefully used in case if I had to organise other articles or items that could consist not just of 1 big part, but would have also other smaller parts. Other examples of use could be several book chapters, or whitepaper parts that could be filed under one reference, but have several parts easily accessible.

**Comment by jlegewie** [comment]
Makes sense and shouldn't be too hard so I will label it as an enhancement. Probably won't get to it anytime soon though

**Comment by r0bis** [comment]
No problem and good luck with the zotfile project; it is much needed.

**Comment by bit-hug** [comment]
Is there any update on this feature? I use it in BibDesk for autofiling and it would be very useful in zotfile 😄

**Comment by galantra** [comment]
This feature would be magnificent!
Some books that I have are downloaded in the form of multiple PDFs, separated according to chapter. In Zotero, this creates dozens of attachments like 12345678910-123 (ISBN-chapterpage).

This means I cannot identify the work when I see the filename outside of Zotero (e.g. DocFetcher), unless I remember part of the ISBN.

I've thought of a workaround: Select the most important books, rename the attached PDFs with an external tool, and modify the filenames in Zotero via the Zutilo addon.

I've also considered creating a custom wildcard. I looked through https://api.zotero.org/itemFields?pprint=1 and it seems like it cannot be done because the filename is not a field. Also, it lists only the fields of _items_, not of attachments (can attachments have fields? Not sure).

**Comment by soumitrakp** [comment]
I would like to have this requested feature.

**Comment by jleaves** [comment]
Any update on this? It'd be so great to have this feature.
